framework/auth/views.py

```diff
@@ -36,95 +36,71 @@
-    verification_key = kwargs['verification_key']
-
-    # Check if request bears a valid verification_key
-    user_obj = get_user(verification_key=verification_key)
-    if not user_obj:
-        error_data = {
-            'message_short': 'Invalid url.',
-            'message_long': 'The verification key in the URL is invalid or has expired.'
-        }
-        raise HTTPError(400, data=error_data)
+    # refresh the verification key (v2)
+    user_obj.verification_key_v2 = generate_verification_key(verification_type='password')
+    user_obj.save()
 
     return {
-        'verification_key': verification_key
+        'uid': user_obj._id,
+        'token': user_obj.verification_key_v2['token'],
     }
 
 
-@collect_auth
-def forgot_password_get(auth, **kwargs):
-    """
-    View to user to land on forgot password page.
-    HTTP Method: GET
-    """
-
-    # If user is already logged in, redirect to dashboard page.
-    if auth.logged_in:
-        return redirect(web_url_for('dashboard'))
-
-    return {}
-
-
-@collect_auth
-def reset_password_post(auth, verification_key=None, **kwargs):
+def reset_password_post(uid=None, token=None):
     """
     View for user to submit reset password form.
     HTTP Method: POST
+
+    :param uid: the user id
+    :param token: the token in verification key
+    :return:
     :raises: HTTPError(http.BAD_REQUEST) if verification_key is invalid
     """
 
-    # If user is already logged in, log user out
-    if auth.logged_in:
-        return auth_logout(redirect_url=request.url)
-
     form = ResetPasswordForm(request.form)
 
-    # Check if request bears a valid verification_key
-    user_obj = get_user(verification_key=verification_key)
-    if not user_obj:
+    # Check if request bears a valid pair of `uid` and `token`
+    user_obj = User.load(uid)
+    if not (user_obj and user_obj.verify_password_token(token=token)):
         error_data = {
-            'message_short': 'Invalid url.',
-            'message_long': 'The verification key in the URL is invalid or has expired.'
+            'message_short': 'Invalid Request.',
+            'message_long': 'The requested URL is invalid, has expired, or was already used',
         }
-        raise HTTPError(400, data=error_data)
+        raise HTTPError(http.BAD_REQUEST, data=error_data)
 
-    if form.validate():
-        # new random verification key, allows CAS to authenticate the user w/o password, one-time only.
-        # this overwrite also invalidates the verification key generated by forgot_password_post
-        user_obj.verification_key = generate_verification_key()
+    if not form.validate():
+        # Don't go anywhere
+        forms.push_errors_to_status(form.errors)
+    else:
+        # clear verification key (v2)
+        user_obj.verification_key_v2 = {}
+        # new verification key (v1) for CAS
+        user_obj.verification_key = generate_verification_key(verification_type=None)
```

Comment:
cas login with username and verification key still uses v1 given it is instant verification